OneCommunityGlobal/HighestGoodNetworkApp
# Jingyi permission management create/edit/delete badge

# Description
This PR is related to badge [permission management fixes](https://docs.google.com/spreadsheets/d/1TfiJY9OLDZuyP2UgF0C1wI9tOSLKbS9MiJFH6n1Dsao/edit#gid=0). It enables users with create/edit/delete badges permission to see the changes of badges immediately.

## Related PRS (if any):
This frontend PR is related to the development backend PR.

## Main changes explained:
Whenever any of the badge management permissions ("createBadges," "updateBadges," or "deleteBadges") are activated for a user, the "seeBadges" permission is automatically included.

## How to test:
1. check into the current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. Clear site data/cache
4. log as admin/owner user
5. go to Other Links -> Permissions Management -> Manage User Permissions -> Badge Permission -> Choose a user without badge management permissions such as 'create badges' 'edit badges' or 'delete badges' -> add the permission for this user -> Scroll down then submit
6. When adding the permission to create/edit/delete the badge, you will find 'see badge' is automatically activated.
7. log into the account that was just granted permission and go to Other Links -> Badge Management to test the permission you just added. For example, create one badge, edit one badge or delete one badge.
8. You will see the change immediately on the page.

## Screenshots or videos of changes:
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/112589476/d879dabd-4d4e-48d4-a60c-7982e0ceb097
